src/actions/transfer.ts

```diff
@@ -7,29 +7,75 @@ import {
   type State,
   parseKeyValueXml,
   composePromptFromState,
-} from '@elizaos/core';
-import { type Hex, formatEther, parseEther } from 'viem';
-
-import { type WalletProvider, initWalletProvider } from '../providers/wallet';
-import { transferTemplate } from '../templates';
-import type { Transaction, TransferParams } from '../types';
+} from "@elizaos/core";
+import { type Hex, encodeFunctionData, formatEther, parseEther } from "viem";
+
+import { type WalletProvider, initWalletProvider } from "../providers/wallet";
+import { transferTemplate } from "../templates";
+import type { Transaction, TransferParams } from "../types";
+
+export const ABIEncoding = async(amount: string, recipientAddress: string, tokenDecimals: string) => {
+  const abi = [
+    {
+        "constant": false,
+        "inputs": [
+            {
+                "name": "_sireId",
+                "type": "address"
+            },
+            {
+                "name": "_matronId",
+                "type": "uint256"
+            }
+        ],
+        "name": "transfer",
+        "outputs": [],
+        "payable": true,
+        "stateMutability": "nonpayable",
+        "type": "function"
+    }
+    ]
+
+  let data = ''
+  
+  if (amount === '0') {
+    data = encodeFunctionData({
+      abi: abi,
+      functionName: 'transfer',
+      args: [
+        recipientAddress,
+        tokenDecimals,
+      ]
+    }) as `0x${string}`
```

Comment:
```suggestion
    }) as Hex
```

You may have to import `Hex`